### DISCUSSION TITLE
Unable to add repos to a github app using github rest api using Fine grained token

### DISCUSSION THREAD

**Discussion by githubofkrishnadhas** [community/community#120984] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hello Team,

I am trying to write an automation to add repositories to a GitHub app to give it to teams to add repos as a self service by setting it as a workflow or as an action. i am using python language to do so.

This is the endpoint: https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#add-a-repository-to-an-app-installation

I have created the Fine grained PAT token for authentication. but the documentation is not clear as it says we cannot use fine grained PAT to add repos to a GitHub app and at same time controversial statement **Fine-grained access tokens for "Add a repository to an app installation"
_This endpoint does not work with GitHub App user access tokens, GitHub App installation access tokens, or fine-grained personal access tokens._** **https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#add-a-repository-to-an-app-installation**  

End point: https://api.github.com/user/installations/{github_app_installation_id}/repositories/{repo_id}
Error code: 404 -->  Resource not found
I am totally stuck and confused. Kindly provide clarification on what needs to be done here.

**Comment by sharoon858** [comment]
responnse #1

**Reply by sharoon2256** [reply]
done

**Comment by githubofkrishnadhas** [comment]
Hi @sharoon2256 @sharoon858  I am looking for how can i fix the issue as the corresponding readme is not clear. 

it would be really help ful if you can lend a hand here 

Thanks!

**Reply by sharoon858** [reply]
What help do you want

**Reply by sharoon858** [reply]
What problem you are facing

**Reply by sharoon858** [reply]
What problem you are facing

**Comment by githubofkrishnadhas** [comment]
As i mentioned above, i am trying to add repos to github app installations using this end point.
endpoint: https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#add-a-repository-to-an-app-installation

![image](https://github.com/community/community/assets/108367225/3956ae62-6bbc-4424-8b7e-1d1ca98dd6da)

I am not sure what credential i can use as GitHub App user access tokens, GitHub App installation access tokens, or fine-grained personal access tokens wont work and this doesnt specify which type of cred/authentication will work as well. if possible can you share what needs to be done. 

Any help is really appreciated @sharoon858 

Thanks!

**Comment by githubofkrishnadhas** [comment, marked as answer]
I have figured out that personal access token classic is required with repo scope permissions.
